README.md

```diff
@@ -9,21 +9,151 @@ Yargs' slack-bot Pirate Joe, the eternal optimist.
 
 ![Joe was one optimistic pirate.](http://i.imgur.com/4WFGVJ9.png)
 
-## Building Slack a Slack-Bot with Yargs
+## Building Slack a Slack-Bot with yargs
 
-With its new headless `parse()` command, and its powerful command
-parsing functionality, yargs is a great tool for building a chat-bot.
+yargs, with its new [headless `parse()`](https://github.com/yargs/yargs#parseargs-context-parsecallback) feature and its powerful command
+parsing functionality, is a great tool for building a chat-bot.
 
-This sample-code and tutorial provides the foundation you need to get a
-chat-bot up in running in Slack using yargs.
+The [Pirate Joe sample project](https://github.com/yargs/pirate-joe/blob/master/server.js) and accompanying tutorial provide the foundation you need to get a chat-bot up in running in Slack using yargs.
 
 ### Parsing Incoming Messages
 
-### Writing Commands
+We've implemented Pirate Joe as a [Slack slash-command](https://api.slack.com/slash-commands). When
+a user types `/joe` in any of yargs' slack channels, a webhook is posted to whatever URL you configure.
 
-### Registring Your Application
+To process this webhook we created a [minimal express application](https://expressjs.com/en/starter/hello-world.html), with the the following `post` handler:
 
-### A Word About Heroku
+```js
+const bodyParser = require('body-parser')
+const express = require('express')
+
+// Slack's slash commands are passed as an urlencoded
+// HTTP post: https://api.slack.com/slash-commands
+app.use(bodyParser.urlencoded({ extended: false }))
+
+// slack webhook endpoint.
+app.post('/', function (req, res) {
+  let context = Object.assign({}, req.body)
+
+  // slack secret token must be provided.
+  if (!req.body || req.body.token !== process.env.SLACK_TOKEN) {
+    return res.sendStatus(401)
+  }
+
+  // provides a respond function that any yargs
+  // command can use to respond to Slack.
+  context.respond = buildResponder(req.body.response_url)
+
+  // run the yargs parser on the inbound slack command.
+  parser.parse(req.body.text || '', context, (err, argv, output) => {
+    if (err) logger.error(err.message)
+    if (output) argv.respond(output)
+  })
+
+  res.send('')
+})
+```
+
+`bodyParser` processes the inbound webhook from Slack and populates three
+variables that are important to us:
+
+* **req.body.token:** the shared secret between our application and
+  and Slack. This allows us to verify that the post originated from Slack.
```

Comment:
Maybe add a note that this token should never be published in eg a repo and should stay an env variable, in the past it happened a lot that people published tokens in repos because they didn't know it's not just verification, but rather authentication